force-app/main/default/classes/HealthCheckVMapper.cls

```diff
@@ -58,54 +58,114 @@ public virtual with sharing class HealthCheckVMapper {
-            
+            List<HealthCheckDefinitionModel> healthCheckDefinitionList = this.getHealthCheckDefinitionList();
+
             for (HealthCheckDefinitionModel healthCheckDefinitionModel : healthCheckDefinitionList) {
                 //Generate view model for current health check definition model
-                HealthCheckDefinitionVModel healthCheckDefinitionVModel = new HealthCheckDefinitionVModel(healthCheckDefinitionModel.name,
-                                                                                                    healthCheckDefinitionModel.className,
-                                                                                                    healthCheckDefinitionModel.namespace);
+                HealthCheckDefinitionVModel healthCheckDefinitionVModel = new HealthCheckDefinitionVModel(
+                    healthCheckDefinitionModel.name,
+                    healthCheckDefinitionModel.className,
+                    healthCheckDefinitionModel.namespace
+                );
 
                 healthCheckDefViewModelsList.add(healthCheckDefinitionVModel);
             }
 
             Integer numberOfRuns = healthCheckDefinitionList.size();
 
             return new HealthCheckVModel(lastRunDate, numberOfRuns, healthCheckDefViewModelsList);
-        }
-        catch (FeatureParameterMapper.FeatureParameterMapperException e) {
+        } catch (FeatureParameterMapper.FeatureParameterMapperException e) {
             throw new AuraHandledException(Label.stgHealthCheckErrorLastRunDate);
         }
     }
-}
\ No newline at end of file
+
+    private List<HealthCheckDefinitionModel> getHealthCheckDefinitionList() {
+        //Totally faking a product registry model here
+        List<ProductRegistryModel> productRegistryModelList = this.getProductRegistryModels();
+
+        List<HealthCheckDefinitionModel> healthCheckDefinitionModelList = new List<HealthCheckDefinitionModel>();
+
+        for (ProductRegistryModel productRegistryModel : productRegistryModelList) {
+            try {
+                //This can go wrong, so we want to catch for this kind of exception
+                Type classType = Type.forName(productRegistryModel.namespace, productRegistryModel.classname);
+
+                //This can *also* go wrong!
+                String productHealthCheckDefinitionJSON = (String) ((Callable) classType.newInstance())
+                    .call(
+                        'Health Check Definition',
+                        new Map<String, Object>{ 'Api Verison' => productRegistryModel.apiVersion }
+                    );
+
+                List<HealthCheckDefinitionModel> healthCheckDefinitionModelListFromProduct = (List<HealthCheckDefinitionModel>) JSON.deserialize(
+                    productHealthCheckDefinitionJSON,
+                    List<HealthCheckDefinitionModel>.class
+                );
```

Comment:
productRegistryModelList.addAll(healthCheckDefinitionModelListFromProduct);

and return productRegistryModelList;

Reply:
That's accurate. Seems I missed that step in POCing it.